### DISCUSSION TITLE
keycloak with oidc

### DISCUSSION THREAD

**Discussion by hanwg** [IgnisDa/ryot#822] [Q&A] [answered]
I'm trying to setup oidc but i'm not getting the prompt to login with oidc.
i've also referred to discussion #756  and verified that my oidc discovery endpoint is valid:
https://keycloak._mydomain_/realms/master/.well-known/openid-configuration.

what could be the issue?
i dont see any errors in the logs.

environment:
- FRONTEND_URL="https://ryot._mydomain_"
- SERVER_OIDC_CLIENT_ID="ryot"
- SERVER_OIDC_CLIENT_SECRET="_secret_"
- SERVER_OIDC_ISSUER_URL="https://keycloak._mydomain_/realms/master/"

**Comment by IgnisDa** [comment]
Can you show me the output of ` https://keycloak.mydomain/realms/master/.well-known/openid-configuration`? You can redact the sensitive parts.

**Reply by hanwg** [reply]
please see attached.
[openid-configuration.json](https://github.com/IgnisDa/ryot/files/15248325/openid-configuration.json)

**Comment by IgnisDa** [comment]
@hanwg I released `v5.2.5` with additional logging when creating the OIDC client. Please upgrade and restart your server and show me the log.

**Reply by hanwg** [reply]
please see attached log.
[log.txt](https://github.com/IgnisDa/ryot/files/15255991/log.txt)

**Reply by IgnisDa** [reply]
Also go to `https://ryot.yourdomain/backend/config` and paste the JSON here. All secrets are redacted so you can directly paste it here.

**Reply by IgnisDa** [reply]
And try setting it to `https://keycloak.mydomain/realms/master` (remove trailing slash).

**Reply by hanwg** [reply]
same issue.
logs are also the same.

**Reply by IgnisDa** [reply]
I updated the logging. The docker image should be published in an hour. Do the upgrade and please LMK the log that shows up in the docker logs.

**Reply by hanwg** [reply, marked as answer]
i managed to get it to work. your commits gave me a hint on what to look out for.
it was my frontend url environment variable:
FRONTEND_URL="https://ryot.mydomain"

removing the double quotes fixed the issue for me.
perhaps you can update the documentation: https://docs.ryot.io/guides/openid
i think you need to quote the entire key-value and not just the value.
e.g. this should work:
"FRONTEND_URL=https://docs.ryot.io/guides/openid"

or dont use quotes at all like what i did.

**Reply by IgnisDa** [reply]
Thanks. Will update the docs.
